cmd/node/node_test.go

```diff
@@ -426,49 +426,28 @@ func TestSpacemeshApp_GrpcService(t *testing.T) {
 	r.NoError(err)
 	r.Equal(false, app.Config.API.StartNodeService)
 
-	const message = "Hello World"
-
-	var conn *grpc.ClientConn
-	require.Eventually(t, func() bool {
-		var err error
-		conn, err = grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()))
-		return err == nil
-	}, 2*time.Second, 100*time.Millisecond)
-
-	r.NoError(err)
-	c := pb.NewNodeServiceClient(conn)
-
-	// We expect this one to fail
-	_, err = c.Echo(context.Background(), &pb.EchoRequest{
-		Msg: &pb.SimpleString{Value: message},
-	})
-	r.Error(err)
-	r.Contains(err.Error(), "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp")
-	r.NoError(conn.Close())
+	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock(), grpc.WithTimeout(3*time.Second))
+	r.ErrorContains(err, "context deadline exceeded")
```

Comment:
This check verifies it there hadn't been already _something_ leasing on that port. In this case 3s timeout seems excesive (as the test will wait for 3 seconds on **happy path**). It will increase run time of the test suite. Perhaps you could shrink it down to, let's say 100ms?

Reply:
Yes, before the changes a lazy connection was opened and then initialized a node client with that. The assertion then checked if sending the first message failed because the then triggered grpc dialup failed.

So the behaviour of the test is still the same: if something is listening on `port` (1242 here) the test fails. Independently of if this is a spacemesh node or something else.

Arguably this check can be removed completely because it doesn't test for correct behaviour of our code but if the environment where it is executed fits restrictions for the test.